ui/store/actions.js

```diff
@@ -673,6 +673,57 @@ const updateMetamaskStateFromBackground = () => {
   });
 };
 
+export function updateSwapApprovalTransaction(txId, txSwapApproval) {
+  return async (dispatch) => {
+    try {
+      await promisifiedBackground.updateSwapApprovalTransaction(
+        txId,
+        txSwapApproval,
+      );
+    } catch (error) {
+      dispatch(txError(error));
```

Comment:
As this codebase is new to me and I am learning, I wonder why `updateTransaction` calls `updateTransactionParams` in the catch statement, and we don't don't call `updateTransactionParams` here.

Reply:
Yeah, the essence of this PR is to only update what is needed and not update everything in a transaction. In this case only the swap settings.

Reply:
Oh got it. Thank you!